VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs

```diff
@@ -1,522 +1,650 @@
+				{
+					Menu.SetChecked($"{CONTROLPOINTS_MENUPATH}/IsSlingshot", false);
+					return false;
+				}
+
+				var dpoint = RetrieveDragPoint(editor.DragPointsEditor, command.userData);
+				if (dpoint != null){
+					Menu.SetChecked($"{CONTROLPOINTS_MENUPATH}/IsSlingshot", dpoint.IsSlingshot);
+				}
+
+				return true;
+			}
+
+			[MenuItem(CONTROLPOINTS_MENUPATH + "/IsSmooth", false, 1)]
+			private static void Smooth(MenuCommand command)
+			{
+				ItemInspector editor = command.context as ItemInspector;
+				if (editor == null || editor.DragPointsEditor == null){
+					return;
+				}
+
+				var dpoint = RetrieveDragPoint(editor.DragPointsEditor, command.userData);
+				if (dpoint != null){
+					editor.DragPointsEditor.PrepareUndo("Changing DragPoint IsSmooth");
+					dpoint.IsSmooth = !dpoint.IsSmooth;
+				}
+			}
+
+			[MenuItem(CONTROLPOINTS_MENUPATH + "/IsSmooth", true)]
+			private static bool SmoothValidate(MenuCommand command)
+			{
+				ItemInspector editor = command.context as ItemInspector;
+				if (editor == null || editor.DragPointsEditor == null || editor.DragPointsEditor.IsItemLocked())
+				{
+					return false;
+				}
+
+				if (!editor.DragPointsEditor.HasDragPointExposition(DragPointExposition.Smooth))
+				{
+					Menu.SetChecked($"{CONTROLPOINTS_MENUPATH}/IsSmooth", false);
+					return false;
+				}
+
+				var dpoint = RetrieveDragPoint(editor.DragPointsEditor, command.userData);
+				if (dpoint != null){
+					Menu.SetChecked($"{CONTROLPOINTS_MENUPATH}/IsSmooth", dpoint.IsSmooth);
+				}
+
+				return true;
+			}
+
+			[MenuItem(CONTROLPOINTS_MENUPATH + "/Remove Point", false, 101)]
+			private static void RemoveDP(MenuCommand command)
+			{
+				ItemInspector editor = command.context as ItemInspector;
+				if (editor == null || editor.DragPointsEditor == null)
+				{
+					return;
+				}
+
+				if (EditorUtility.DisplayDialog("DragPoint Removal", "Are you sure you want to remove this Dragpoint ?", "Yes", "No"))
+				{
+					editor.DragPointsEditor.RemoveDragPoint(command.userData);
+				}
+			}
+
+			[MenuItem(CONTROLPOINTS_MENUPATH + "/Remove Point", true)]
+			private static bool RemoveDPValidate(MenuCommand command)
+			{
+				ItemInspector editor = command.context as ItemInspector;
+				if (editor == null || editor.DragPointsEditor == null || editor.DragPointsEditor.IsItemLocked())
+				{
+					return false;
+				}
+
+				return true;
+			}
+
+			//Curve Traveller
+			[MenuItem(CURVETRAVELLER_MENUPATH + "/Add Point")]
+			static void AddDP(MenuCommand command)
+			{
+				ItemInspector editor = command.context as ItemInspector;
+				if (editor == null || editor.DragPointsEditor == null)
+				{
+					return;
+				}
+
+				editor.DragPointsEditor.AddDragPointOnTraveller();
+			}
+		}
+
+		public bool IsItemLocked()
+		{
+			IEditableItemBehavior editable = _target as IEditableItemBehavior;
+			if (editable == null)
+			{
+				return true;
+			}
+			return editable.IsLocked;
+		}
```

Comment:
Could be a oneliner:

```cs
private bool IsItemLocked => !(_target is IEditableItemBehavior editable) || editable.IsLocked;
```

Reply:
didn't know :) will change

Reply:
This is still open, right? It's just in `DragPointsItemInspector` now?